include/fastdds/dds/xtypes/dynamic_types/DynamicData.hpp

```diff
@@ -0,0 +1,1041 @@
+     */
+    virtual uint32_t get_item_count() = 0;
+
+    /*!
+     * Clear all members associated to the object.
+     * @return @ref ReturnCode_t
+     * @retval RETCODE_OK when the cleaning was successful.
+     */
+    virtual ReturnCode_t clear_all_values() = 0;
+
+    /*!
+     * Clear all members not associated to the key
+     * @return @ref ReturnCode_t
+     * @retval RETCODE_OK when the cleaning was successful.
+     */
+    virtual ReturnCode_t clear_nonkey_values() = 0;
+
+    /*!
+     * Clear a member.
+     * The meaning of "clearing" depends on the type of the sample:
+     * @li If aggregated type, and the indicated member is optional, remove it.
+     * If the indicated member is not optional, set it to its default value.
+     * @li If variable-length collection type, remove the indicated element, shifting any subsequence elements to the
+     * next-lowest index.
+     * @li If array type, set the indicated element to its default value.
+     * @li If bitmask type, clear the indicated bit.
+     * @li If enumerated type, set it to the first value of the enumerated type.
+     * @li If primitive type, set it to its default value.
+     * @param[in] id Identifier of the member to purge
+     * @return @ref ReturnCode_t
+     * @retval RETCODE_OK when the cleaning was successful.
+     * @retval RETCODE_BAD_PARAMETER when the @ref MemberId is invalid.
+     */
+    virtual ReturnCode_t clear_value(
+            MemberId id) = 0;
+
+    /*!
+     * \b Loans a @ref DynamicData reference within the sample
+     * @remarks This loan shall be returned by the @ref DynamicData::return_loaned_value operation
+     * @param[in] id identifier of the object to retrieve
+     * @return @ref DynamicData reference loaned or \b nil on outstanding loaned data
+     */
+    virtual traits<DynamicData>::ref_type loan_value(
+            MemberId id) = 0;
+
+    /*!
+     * Returns a loan retrieved using @ref DynamicData::loan_value.
+     * @param[in] value @ref DynamicData reference previously loaned
+     * @retval RETCODE_OK when the loaned was returned successfully.
+     * @retval RETCODE_PRECONDITION_NOT_MET when the loan is invalid.
+     */
+    virtual ReturnCode_t return_loaned_value(
+            traits<DynamicData>::ref_type value) = 0;
+
+    /*!
+     * Creates and returns a new data sample with the same contents as this one.
+     * A comparison of this object and the clone using equals immediately following this call will return `true`.
+     * @return @ref DynamicData reference
+     */
+    virtual traits<DynamicData>::ref_type clone() = 0;
+
+    /*!
+     * Retrieves an \b int32 value associated to an identifier.
+     * @param[inout] value \b int32 to populate
+     * @param[in] Id identifier of the member to query.
+     * @return @ref ReturnCode_t
+     * @retval RETCODE_OK when the value was retrieved successfully.
+     * @retval RETCODE_BAD_PARAMETER when the @ref MemberId is invalid or the member type is not \b int32.
```

Comment:
According to section 7.5.2.11.9, there are implicit type promotions allowed. This suggestion applies to other similar comments in the methods below.

```suggestion
     * @retval RETCODE_BAD_PARAMETER when the @ref MemberId is invalid or the member type is not promotable to \b int32.
```

Reply:
Done in 9be524d12

Reply:
This has been applied only to one of the methods. There are a lot of methods which this suggestion also applies